### ISSUE TITLE
TGSM race conditions are flagged as errors

### ISSUE THREAD

**Issue by dmpots** [Microsoft/DirectXShaderCompiler#65]
In commit #48 we implemented checking for race conditions, but these are flagged as errors. I think the old compiler flagged these as warnings. We should keep that same behavior.

compiling the below program gives the error:

    error: validation errors
    t.hlsl:21:15 Race condition writing to shared memory detected, consider making this write conditional

```
// RUN: dxc /Tcs_5_1 /DDX12  %s

struct Foo {
    int a;
    int b;
    int c;
    int d;
};

Buffer<Foo> inputs : register(t1);
RWBuffer< int > g_Intensities : register(u1);

groupshared Foo sharedData;

#ifdef DX12
[RootSignature("DescriptorTable(UAV(u1, numDescriptors=1), SRV(t1), visibility=SHADER_VISIBILITY_ALL)")]
#endif
[ numthreads( 64, 2, 2 ) ]
void main( uint GI : SV_GroupIndex)
{
	sharedData = inputs[GI];
	int rtn;
	InterlockedAdd(sharedData.d, g_Intensities[GI], rtn);
	g_Intensities[GI] = rtn + sharedData.d;
}

```

**Comment by python3kgae** [comment]
The check is disabled for now.
